src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Application {
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        for(int i = 0; i < figures.length/2; i++) {
+            FigureSupplier figureSupplier = new FigureSupplier();
```

Comment:
The FigureSupplier does not need to be created every time in the loop. Create once before the loop and use inside the loop